map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java

```diff
@@ -101,60 +93,50 @@ public void start(Stage stage) {
+        // if there is an error or warning, format and display a message as an Alert
         ArcGISRuntimeException error = statusChangeEvent.getError();
         if (error != null) {
           Throwable cause = error.getCause();
-          String message = (cause != null) ? cause.toString() : error.toString();
+          String message = (cause != null) ? "Network issue, unable to load layer" : error.getMessage();
           Alert alert = new Alert(Alert.AlertType.ERROR, message);
           alert.show();
         }
       });
 
       // create buttons to toggle the visibility of the feature layer
-      loadLayerButton = new Button("Load Layer");
-      loadLayerButton.getStyleClass().add("panel-button");
-      hideLayerButton = new Button("Hide Layer");
-      hideLayerButton.getStyleClass().add("panel-button");
-      // initially add load layer button to the control panel
-      controlsVBox.getChildren().add(loadLayerButton);
+      toggleLayerButton = new Button("Load Layer");
 
       // create a listener for clicks on the load layer button
-      loadLayerButton.setOnAction(event -> {
+      toggleLayerButton.setOnAction(event -> {
 
-        // if the feature layer already exists and is hidden, toggle it's visibility to visible
-        if (featureLayer != null && !featureLayer.isVisible()) {
-          featureLayer.setVisible(true);
+        // if the feature layer already exists toggle it's visibility to visible
+        if (featureLayer != null) {
+          featureLayer.setVisible(!featureLayer.isVisible());
         } else {
-          // create a feature layer from a portal item
+          // if the feature layer doesn't exist, create a new feature layer from a portal item
           final PortalItem portalItem = new PortalItem(new Portal("https://runtime.maps.arcgis.com/"),
             "b8f4033069f141729ffb298b7418b653");
           featureLayer = new FeatureLayer(portalItem, 0);
           //  set a minimum and a maximum scale for the visibility of the feature layer
           featureLayer.setMinScale(40000000);
-          featureLayer.setMaxScale(40000000 / 10);
+          featureLayer.setMaxScale(4000000);
           // add the feature layer to the map
           map.getOperationalLayers().add(featureLayer);
         }
 
-        // toggle the visibility of the buttons
-        displayHideButton();
-      });
-
-      // create a listener for clicks on the hide layer button to toggle visibility of the feature layer
-      hideLayerButton.setOnAction(event -> {
-
-        if (featureLayer == null) return;
-
-        if (featureLayer.isVisible()) {
-          featureLayer.setVisible(false);
-          displayLoadButton();
+        // toggle the text on the button
+        if (toggleLayerButton.getText().equals("Load Layer")){
+          toggleLayerButton.setText("Hide Layer");
         } else {
-          featureLayer.setVisible(true);
-          displayHideButton();
+          toggleLayerButton.setText("Load Layer");
```

Comment:
Show this now be `show` as the opposite of `hide`? We're only really 'loading' the layer at the very first click, and then just showing/hiding. Unless I'm misunderstanding?

```suggestion
          toggleLayerButton.setText("Show Layer");
```

Reply:
Yes good point, now the button implementation has changed it is simple to have Load initially, then show/hide. I tweaked the if statement in the `toggleLayerButton` listener and it works well, thanks for this suggestion!